shell2http.go

```diff
@@ -576,6 +576,21 @@ func errChainAll(chainFuncs ...func() error) error {
 	return resErr
 }
 
+// mwLimitHeaderLength - limits the length of headers in the request
+func mwLimitHeaderLength(next http.HandlerFunc, maxHeaderLength int) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
```

Comment:
here we have already read the entire request into memory, and stdlib http-server already have limit for header size, see `DefaultMaxHeaderBytes` in https://pkg.go.dev/net/http, the default limit of 1MB seems to be sufficient